src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -140,11 +140,12 @@ private static string ParseHostNameFromHeader(string hostHeader)
         private static HttpConnectionKey GetConnectionKey(HttpRequestMessage request, Uri proxyUri, bool isProxyConnect)
         {
             Uri uri = request.RequestUri;
+            string host = uri.HostNameType == UriHostNameType.IPv6 ? "[" + uri.IdnHost + "]" : uri.IdnHost;
```

Comment:
This is adding an allocation for every request using an IPv6 address as a host name.  Is that necessary?  Since the resulting HttpConnectionKey is used to choose the pool to use, can't this be cached on the pool instead?  And if not, can we skip doing this if proxyUri == null?

Reply:
> can't this be cached on the pool instead?

I think it's possible. I didn't do that in the first place, because we will need a helper logic to determine if the address we passed in is IPv6 or not (we pass in string, not uri), and I thought the helper logic could result in additional allocation. I will try to explore this option, if not able to come up with efficient way, we may need to try the second appraoch -> skip doing this if proxyUri == null.

Reply:
IPv6 host names are unusual, saving a few extra allocations for them is not a high priority.